packages/angular_devkit/core/src/json/schema/utility.ts

```diff
@@ -23,6 +24,30 @@ export function getTypesOfSchema(schema: JsonObject | true): Set<string> {
     potentials = new Set([schema.type]);
   } else if (Array.isArray(schema.type)) {
     potentials = new Set(schema.type as string[]);
+  } else if (isJsonArray(schema.enum)) {
+    // Gather the type of each enum values, and use that as a starter for potential types.
+    const filteredTypes = [...schema.enum].map(v => {
```

Comment:
for/of would probably be cleaner here and reduce the iterations from four to one.

Reply:
Done. It does look cleaner.